test/runtests.jl

```diff
@@ -472,3 +473,95 @@ end
 
 c = ConstrainedFieldV1(field = "1")
 @test c.field == 1
+
+#####
+##### record error reporting
+#####
+
+@schema "test.field-error" FieldError
+
+function _validate(x)
+    x in ("a", "b", "c") || throw(ArgumentError("Must be a, b, or c"))
+    return x
+end
+
+@version FieldErrorV1 begin
+    a::Union{String,Missing} = Legolas.lift(_validate, a)
+    b::(<:Union{String,Missing}) = Legolas.lift(_validate, b)
+    c::Union{Integer,Missing}
+    d::(<:Union{Integer,Missing})
+end
+
+_num_calls = Ref{Int}(0)
+@version FieldErrorV2 begin
+    a::Integer = begin
+        _num_calls[] += 1
+        a isa Function ? a() : a
+    end
+end
+
+@version FieldErrorV3 begin
+    a::Integer = replace(a, ' ' => '-')
+end
+
+@testset "Legolas record constructor error handling" begin
+    @testset "field constructor error" begin
+        ex_stack = try
+            FieldErrorV1(; a="invalid")
+        catch
+            current_exceptions()
+        end
+
+        @test length(ex_stack) == 2
+        @test sprint(showerror, ex_stack[1].exception) == "ArgumentError: Must be a, b, or c"
+        @test sprint(showerror, ex_stack[2].exception) == "ArgumentError: Invalid value set for field a (\"invalid\")"
```

Comment:
this is so nitpicky but can we make the error more clearly set the field name apart?  like `` `a` `` or `:a` or something like that?

Reply:
Ended up chatting about this and we'll go with double quotes. Will call out that the `TypeError` which we also use typically doesn't quote things which is kind of strange:
```julia
julia> Rational{String}
ERROR: TypeError: in Rational, in T, expected T<:Integer, got Type{String}
```

Reply:
As I was adding the quoting I just noticed Legolas uses backticks for other error messages referring to fields so we'll just stick with that convention